src/BasicAuthRequest.jl

```diff
@@ -0,0 +1,36 @@
+module BasicAuthRequest
+
+if VERSION > v"0.7.0-DEV.2338"
+using Base64
+end
+
+import ..Layer, ..request
+using ..URIs
+using ..Pairs: getkv, setkv
+import ..@debug, ..DEBUG_LEVEL
+
+
+"""
+    request(BasicAuthLayer, method, ::URI, headers, body) -> HTTP.Response
```

Comment:
I'd like to have all the docs for the layers be `HTTP.request(...` to encourage prefixing the call w/ HTTP. Even if `request` is exported, given the current state of potential function name clashes in Julia (i.e. there's no cross-module method merging yet), it's only wise and safe to always prefix.

Reply:
The top level HTTP.jl docs have the prefix.
My reasoning here was that the per-layer docs are internal architecture doc, not user API doc.
I agree with encouraging the prefix. I think probably HTTP.jl should not export any new symbols into the global namespace.

Reply:
sounds good